### ISSUE TITLE
3DS2

### ISSUE THREAD

**Issue by coxy** [braintree/braintree-web#424]
Hello,

I'm following the 3DS2 guide here: https://developers.braintreepayments.com/guides/3d-secure/migration/javascript/v3 since you "recommend integrating 3DS 2.0 by April 14, 2019, to meet card brand guidelines."

The document references the following version of the three-d-secure module: https://js.braintreegateway.com/web/3.43.0-beta.4/js/three-d-secure.min.js

I noticed the current npm release is at 3.44.1, however none of the 3DS2 updates appear to be in this release.

Please let me know when these updates will be integrated here and published on npm as the deadline is approaching.

Thanks.

**Comment by siculo** [comment]
Hi,

I'm interested in this topic too, because supposedly there is no 3DS2 support available in braintree-web.

**Comment by adorum** [comment]
I am getting error `Cannot read property 'cardinalAuthenticationJWT' of null`when calling 
```
threeDSecure.create({
  client,
  version: 2,
}, (err, threeDSecureInstance) => {
  console.log(err);
});
```

**Comment by crookedneighbor** [comment]
Sorry for being late on this.

@coxy The April 14th deadline was one set by the card brands, but none of them actually implemented 3ds 2.0 by that deadline. Our migration guide will be updated soon to reflect that. When we have more info, we'll update here as well.

@siculo We're hoping to release a new beta today that is up to date with the changes that have been made on master

@adorum I'm addressing your issue in the issue you opened.

**Comment by AnuraagBasu** [comment]
Hello,

I'm following the 3DS 2.0 adoption guide(https://developers.braintreepayments.com/guides/3d-secure/migration/javascript/v3) to migrate our payment flow, using version 3.45.0 of braintree-web.

I'm using the _hostedFields_ object from the module for displaying and tokenizing the card details.
But, I can't seem to find the _verifyCard()_ method on the hostedFieldsInstance. Hence, I'm not able to complete the last step of the migration docs.

Please someone help me out here.

**Comment by crookedneighbor** [comment]
`verifyCard` is not on the hosted fields instance. It is on the 3D Secure component instance.

**Comment by crookedneighbor** [comment]
https://developers.braintreepayments.com/guides/3d-secure/migration/javascript/v3#configure-3ds-version

**Comment by AnuraagBasu** [comment]
Hi @crookedneighbor,
Does that mean I do not need to call _verifyCard_ if I'm using _hostedFields_?
If I do not use _hostedFields_, how should I tokenize the card details to generate a nonce?

**Comment by crookedneighbor** [comment]
@AnuraagBasu I think you're best bet is going to be to reach out to our support team: https://help.braintreepayments.com/

They'll be equipped to help you with your integration and answer your questions.

**Comment by crookedneighbor** [comment]
3DS v2 is now available in 3.47.0
